src/Parsing/Impl/FStringParser.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.Diagnostics;
+using Microsoft.Python.Core.Text;
+using Microsoft.Python.Parsing.Ast;
+
+namespace Microsoft.Python.Parsing {
+    public class FStringParser {
+        private readonly string _fString;
+        private readonly StringBuilder _buffer = new StringBuilder();
+        private readonly FStringBuilder _builder;
+        private readonly ErrorSink _errors;
+        private int _errorCode = 0;
+        private int _position = 0;
+
+        public FStringParser(FStringBuilder builder, string fString, ErrorSink errors) {
```

Comment:
Is `FStringBuilder` ever used outside of `FStringParser`? If not, `FStringParser` can just `new` its own builder